src/reducers/workspaces.ts

```diff
@@ -339,6 +340,7 @@ export const reducer: Reducer<IWorkspaceManagerState> = (
         }
       };
     case EVAL_TESTCASE_SUCCESS:
+      const testcaseSuccessCopy = cloneDeep(state[location].editorTestcases);
```

Comment:
May I know why deep copy is used?

Reply:
The current implementation seems to be modifying the original editorTestcases in the test file. After some discussion with @rrtheonlyone , we decided to use this implementation.

Reply:
Done